container/test-impl-base/src/main/java/org/jboss/arquillian/container/test/impl/client/deployment/DeploymentGenerator.java

```diff
@@ -169,6 +170,7 @@ private void buildTestableDeployments(DeploymentScenario scenario, TestClass tes
                 if (ClassContainer.class.isInstance(applicationArchive)) {
                     ClassContainer<?> classContainer = ClassContainer.class.cast(applicationArchive);
                     classContainer.addClass(testCase.getJavaClass());
+                    addAdditionalObserverClass(classContainer, testCase.getJavaClass());
```

Comment:
I would either add `IfPresent` to the method name or move `if` outside of the method.